src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -54,18 +57,33 @@ public void checkUserInDatastore(User user) {
     // ** NEED TO WRITE TEST FOR BEHAVIORAL TESTING
 
     Key userKey = KeyFactory.createKey("User", user.getUserId());
-    if (datastore.get(userkey) == null) {
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-
-        // Create new user entity 
-        Entity userEntity = user.createEntity();
-        Key userKey = userEntity.getKey();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity userEntity;
 
-        // Create User's key and add to object 
-        String userKeyStr = KeyFactory.keyToString(userKey);
-        user.setUserKey(userKeyStr);
+    // Check if user is in datastore
+    try {
+      userEntity = datastore.get(userKey);
+      user.setUserKey(KeyFactory.keyToString(userKey));
+    } catch (EntityNotFoundException e) {
+      userEntity = null;
+    }
 
-        datastore.put(userEntity);
+    if (userEntity == null) {
+        addUserToDatastore(datastore, user);
     }
   }
+
+  public void addUserToDatastore(DatastoreService datastore, User user) {
```

Comment:
`storeUserInDatastore`. In this context, store (or a similar word) is a less ambiguous word than add (adding a user to a database could mean creating a new user account with permissions to access the database).

Reply:
Fixed in #5